web/pdf_viewer.js

```diff
@@ -250,6 +250,14 @@ var PDFViewer = (function pdfViewer() {
           });
           self.container.dispatchEvent(event);
         };
+        pageView.textLayerOnAfterDraw =
+            function pdfViewLoadTextLayerOnAfterDraw() {
+          var event = document.createEvent('CustomEvent');
+          event.initCustomEvent('textlayerrendered', true, true, {
+            pageNumber: pageView.id
+          });
+          self.container.dispatchEvent(event);
```

Comment:
I think you can move event dispatching to the text_layer_builder.js (and avoid callback stuff)